### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by Dunstable**

### CODE REVIEW

**Comment by janko-isidorovic** [comment]
Hi Dunstable, thanks for your contribution. Please, sign-off this pull request.

**Comment by Dunstable** [comment]
The process states, "At least one approved review is required by reviewers with write access."

**Comment by trcox** [comment]
@Dunstable, you can read more about the sign-off check [https://github.com/probot/dco#how-it-works](here). We require all contributions to EdgeX Foundry to be signed off on by their contributor.

To sign off on a previous commit, run:

> git commit --amend --sign-off

Then update your pull request by overwriting the previous commit with a force push.

File: src/main/java/org/edgexfoundry/messaging/MQTTSender.java [hunk 1]
```diff
@@ -97,6 +101,16 @@ private void connectClient() {
...
 			connOpts.setCleanSession(true);
 			connOpts.setKeepAliveInterval(keepAlive);
+			if (secureConnection) {
+				try {
+					EdgeXSecurityManager secMgr = new EdgeXSecurityManager();
+					SSLContext ctxt = SSLContext.getInstance("TLS");
+					ctxt.init(secMgr.getKeyManagers(), secMgr.getTrustManagers(), new java.security.SecureRandom());
+					connOpts.setSocketFactory(ctxt.getSocketFactory());
+				} catch (Exception e) {
+					logger.debug("Failed to initialize secure connection to broker.  Will attempt unsecure connection.", e);
```
Comment (trcox) [inline src/main/java/org/edgexfoundry/messaging/MQTTSender.java]: Is this our desired failure behavior when attempting an MQTTS connection? I can see arguments in both directions (security vs. functionality), but perhaps this needs to be configurable with some sort of strictness check (@janko-isidorovic @jpwhitemn)?

Regardless, log level for this sort of failure should be bumped up to the `error` level.

**Comment by Dunstable** [comment]
I made changes requested by trcox.  However, the DCO got messed up and I need to find someone who knows how to fix that.

**Comment by darkodraskovic** [comment]
[This](https://stackoverflow.com/a/36351995) is a very good explanation how ammend works. 

However, I personally didn't have much luck with `ammend` and sign-off. For example, when I ammend my latest commit with a sign-off, git recognizes it. However, git pull system wich do the checks still reports that a sign-off is missing.

However, I did sign-off my last commit with a `git rebase --interactive`. [This article](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history) explains well the latter command.

**Comment by Dunstable** [comment]
DCO is fixed and passes now.  Branch history also cleaned up:  errant checkin removed that arose when attempting to fix aforementioned DCO issue.  Thanks for everyone's help!

**Comment by trcox** [comment]
@Dunstable I got in touch with Jim offline, we will want to set the behavior for a failed MQTTS exception to cause a failure to initialize.

File: src/main/java/org/edgexfoundry/messaging/MQTTSender.java [hunk 2]
```diff
@@ -97,6 +101,16 @@ private void connectClient() {
...
 			connOpts.setKeepAliveInterval(keepAlive);
+			if (secureConnection) {
+				try {
+					EdgeXSecurityManager secMgr = new EdgeXSecurityManager();
+					SSLContext ctxt = SSLContext.getInstance("TLS");
+					ctxt.init(secMgr.getKeyManagers(), secMgr.getTrustManagers(), new java.security.SecureRandom());
+					connOpts.setSocketFactory(ctxt.getSocketFactory());
+				} catch (Exception e) {
+					logger.error("Failed to initialize secure connection to broker.  Will attempt unsecure connection.", e);
+				}
```
Comment (trcox) [inline src/main/java/org/edgexfoundry/messaging/MQTTSender.java]: Let's raise an MQTTException here after the error log to fail cleanly on MQTTS initialization failure.

**Pull request merged**